server/src/mds3.init.js

```diff
@@ -1595,7 +1596,34 @@ async function validate_query_rnaseqGeneCount(ds, genome) {
 	also require that there's no duplicate samples in header line, so rust/r won't break
 	*/
 	{
-		const samples = (await getFirstLine(q.file)).trim().split('\t').slice(4)
+		let samples = []
+		if (!ds.queries.rnaseqGeneCount.storage_type || ds.queries.rnaseqGeneCount.storage_type == 'text') {
```

Comment:
shouldn't allow storage_type to be missing

Reply:
The reason its optional, is that then we would have to change all other dataset files. Right now in the absence of `storage_type` in dataset file, its presumed that data is stored in a text file. In the future, we will convert all DE text files to HDF5.

Reply:
please pull sjpp branch and change storage_type into required property